core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -54,6 +55,14 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
 
+        Optional<ConnectorMaterializedViewDefinition> view = metadata.getMaterializedView(session, tableName);
+        if (view.isPresent()) {
```

Comment:
Does `metadata.getTableHandle` return non-empty for materialized view? If not please move the new code to `if (tableHandle.isEmpty()) {`

Reply:
Shouldn't we check for `if (!statement.isExists()) {` here as below?

Reply:
I added guards for both view and materialized view. Hive connector returns view from `getTableHandle` method so it needs to happen even if `getTableHandle` returns a not empty value.
I added case where `IF EXISTS` is used + tests

Reply:
>   // those two guards are here because some connectors violate SPI contract and return views or materialized views from getTableHandle.

Which connectors? Can we fix them? Also, that would argue for detecting misbehaving connectors and failing with some kind of internal error to indicate so.

Reply:
hive connector returns views from `getTableHandle`. This in turn makes `GRANT SELECT ON...` or `COMMENT ON TABLE`  work on views.